primitives/src/predicate.rs

```diff
@@ -114,15 +115,43 @@ pub fn run(rule: &Rule, context: &Context) -> bool {
 // ======================================================
 
 /// It checks the existential of a claim.
-#[derive(Clone, Debug)]
+///
+/// # `CustomerDueDiligence` wildcard search
+/// The `CustomerDueDiligence` claim supports wildcard search if you use the default `CddId` (a zero filled data).
+/// For instance:
+///     - The `exists(Claim::CustomerDueDiligence(CddId::default()))` matches with any CDD claim.
+///     - The `exists(Claim::CustomerDueDiligence(a_valid_cdd_id))` matches only for the given
+///     `a_valid_cdd_id`.
+///
+#[derive(Clone)]
+#[cfg_attr(feature = "std", derive(Debug))]
 pub struct ExistentialPredicate<'a> {
     /// Claims we want to check if it exists in context.
     pub claim: &'a Claim,
 }
 
 impl<'a> Predicate for ExistentialPredicate<'a> {
-    #[inline]
     fn evaluate(&self, context: &Context) -> bool {
+        match &self.claim {
+            Claim::CustomerDueDiligence(ref cdd_id) if !cdd_id.is_valid() => {
```

Comment:
To avoid confusion, I think we should rename `is_valid()` to something like `is_wildcard` or `is_default`.